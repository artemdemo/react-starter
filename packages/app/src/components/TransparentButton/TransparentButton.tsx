import React, { ButtonHTMLAttributes } from 'react';
import classnames from 'classnames';
import s from './TransparentButton.module.css';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {}

export const TransparentButton: React.FC<Props> = (props) => {
  return (
    <button
      {...props}
      className={classnames(s.TransparentButton, props.className)}
    >
      {props.children}
    </button>
  );
};
