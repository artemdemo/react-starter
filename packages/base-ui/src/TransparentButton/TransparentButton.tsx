import React, { ButtonHTMLAttributes } from 'react';
import classnames from 'classnames';
import './TransparentButton.module.css';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {}

export const TransparentButton: React.FC<Props> = (props) => {
  return (
    <button
      {...props}
      className={classnames('TransparentButton', props.className)}
    >
      {props.children}
    </button>
  );
};
