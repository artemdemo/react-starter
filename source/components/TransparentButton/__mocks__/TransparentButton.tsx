import React from 'react';
import _omit from 'lodash/omit';

const TransparentButton = props => (
    <button data-mock='TransparentButton'>
        {JSON.stringify(_omit(props, 'children'), null, 2)}
        {props.children}
    </button>
);

export default TransparentButton;
