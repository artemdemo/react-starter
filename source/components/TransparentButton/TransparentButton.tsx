import React from 'react';
import classnames from 'classnames';
import './TransparentButton.css';

type Props = {
  className?: string;
  onClick?: () => void;
};

export const TransparentButton: React.FC<Props> = (props) => (
  <button className={classnames('TransparentButton', props.className)} onClick={props.onClick}>
    {props.children}
  </button>
);
