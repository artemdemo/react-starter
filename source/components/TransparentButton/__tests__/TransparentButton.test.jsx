import React from "react";
import renderer from "react-test-renderer";
import TransparentButton from "../TransparentButton";

describe('TransparentButton', () => {
    it('render default', () => {
        const tree = renderer.create(
            <TransparentButton>
                Test
            </TransparentButton>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
