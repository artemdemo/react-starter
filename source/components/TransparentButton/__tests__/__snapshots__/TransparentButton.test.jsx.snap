// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransparentButton render default 1`] = `
<button
  className="sc-bdnylx deAxwH"
>
  Test
</button>
`;
